digitaltwin/modelrepo/Authentication/ModelSharedAccessSignatureBuilder.cs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Net;
+
+namespace Microsoft.Azure.Devices.Common.Authorization
+{
+    public class ModelSharedAccessSignatureBuilder : SharedAccessSignatureBuilder
+    {
+        public string RepositoryId { get; set; }
+
+        public override string ToSignature()
+        {
+            return BuildSignatureForModelRepo(KeyName, Key, HostName, RepositoryId, TimeToLive).ToString();
+        }
+
+        public StringBuilder BuildSignatureForModelRepo(string keyName, string key, string Hostname, string repositoryId, TimeSpan timeToLive)
+        {
+            string expiresOn = BuildExpiresOn(TimeToLive);
```

Comment:
Why not call the base class impl, and then append the repo id to the result?

Reply:
That's definitely a great idea.